### PR TITLE
add edmfx buoyancy gradient

### PR DESCRIPTION
**Pull request by szy21**
## Purpose 

Add buoyancy gradient calculation. Only `BuoyGradMean` is added as `BuoyGradQuadratures` is not used.

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/cache/edmf_precomputed_quantities.jl [hunk 1]
```diff
@@ -163,8 +163,35 @@ function set_edmf_precomputed_quantities!(Y, p, ᶠuₕ³, t)
...
+        params,
+        moisture_model,
+        EnvBuoyGrad(
+            BuoyGradMean(),
+            TD.air_temperature(thermo_params, ᶜts⁰),           # t_sat
+            TD.vapor_specific_humidity(thermo_params, ᶜts⁰),   # qv_sat
+            ᶜspecific⁰.q_tot,                                  # qt_sat
+            TD.dry_pottemp(thermo_params, ᶜts⁰),               # θ_sat
+            TD.liquid_ice_pottemp(thermo_params, ᶜts⁰),        # θ_liq_ice_sat
+            ᶜdivᵥ(
```
Comment (szy21) [inline src/cache/edmf_precomputed_quantities.jl:178]: check if we should use norm(grad) instead

**Review by charleskawczynski — COMMENTED** [review]
I'm not a fan of the code duplication, at least the `buoyancy_gradients`, `buoyancy_gradient_chain_rule` and `EnvBuoyGrad` methods-- they can easily be shared between the dycore and the deprecated edmf code.

I think deleting the old code is a matter of: deleting the functions and adding

```julia
import ..buoyancy_gradient_chain_rule
```
at the top of `src/TurbulenceConvection_deprecated/TurbulenceConvection.jl` (similar to other pieces that have been unified).

The other parts (where it's used) like
```julia
@. ᶜ∂b∂z = buoyancy_gradients(...)
```
is fine to duplicate-- this is much more difficult to unify because it requires that cache variable names match up.

If this was copied verbatim and we don't anticipate changing them / names, I would prefer deleting the old methods

**Comment by szy21** [comment]
> If this was copied verbatim and we don't anticipate changing them / names, I would prefer deleting the old methods

I didn't change the old code mostly because I thought we would delete the entire folder after the new EDMF is working. But I am fine with deleting it. @trontrytel What do you think?

**Comment by szy21** [comment]
Actually, @charleskawczynski We anticipate changing the function to calculate the derivative to enthalpy instead of liquid ice potential temperature at some point. But that would be after the old EDMF is deleted I think.

**Comment by szy21** [comment]
@charleskawczynski I just talked with Tapio and Anna. We think we may change these functions in the future, and would like to keep the old EDMF for comparison. In that case, would it be ok to keep the old code?

**Comment by charleskawczynski** [comment]
Yes, that's fine

**Review by charleskawczynski — APPROVED** [review]

File: src/cache/edmf_precomputed_quantities.jl [hunk 2]
```diff
@@ -163,8 +163,35 @@ function set_edmf_precomputed_quantities!(Y, p, ᶠuₕ³, t)
             ᶜbuoyancy(params, ᶜρ_ref, ᶜρ⁰),
         )
     end
+    
+    # First order approximation: Use environmental mean fields.
+    @. ᶜ∂b∂z = buoyancy_gradients(
```
[the remark this answers is not in the document]
Comment (szy21) [reply src/cache/edmf_precomputed_quantities.jl:168]: We changed it to norm(ᶜgradᵥ) and it also works.

[the remark this answers is not in the document]
Comment (szy21) [reply src/cache/edmf_precomputed_quantities.jl:168]: Ah, wait, the sign is not correct with the norm. Let me think.

**Pull request merged**